### PR TITLE
Improve state management

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Improve state management
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | Previous configuration are no longer taken into account, log files are no longer cut during an update

### CODE REVIEW

File: classes/State.php [hunk 1]
```diff
@@ -298,36 +303,44 @@ public function getRestoreVersion(): ?string
...
         return $this;
     }
 
     public function setBackupDbFilename(string $backupDbFilename): State
     {
         $this->backupDbFilename = $backupDbFilename;
+        $this->save();
 
         return $this;
     }
```
Comment (ga-devfront) [inline classes/State.php]: you can change this methods to private method and only save on `setBackupName`

**Review by ga-devfront — CHANGES_REQUESTED** [review]

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by ga-devfront — DISMISSED** [review]

File: classes/Commands/UpdateCommand.php [hunk 2]
```diff
@@ -127,17 +129,17 @@ private function chainCommand(OutputInterface $output): int
             return ExitCode::SUCCESS;
         }
 
-        if (strpos($lastInfo, 'bin/console update:start') !== false) {
+        if (strpos($lastInfo, self::$defaultName) !== false) {
```
Comment (M0rgan01) [inline classes/Commands/UpdateCommand.php:134]: this posed a problem if options were placed before the command (ex: `bin/console -v update:start`)

File: classes/Commands/UpdateCommand.php [hunk 3]
```diff
@@ -83,6 +84,7 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
             if ($isFirstUpdateProcess) {
                 $this->logger->debug('Cleaning previous state files.');
                 $this->upgradeContainer->getFileConfigurationStorage()->cleanAllUpdateFiles();
+                $this->upgradeContainer->getFileConfigurationStorage()->clean(UpgradeFileNames::CONFIG_FILENAME);
```
Comment (M0rgan01) [inline classes/Commands/UpdateCommand.php:92]: We could end up with the config of an old update

**Review by ga-devfront — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Result of the logs folder after an update:

```
root@b653c57bea07:/var/www/html/admin-dev/autoupgrade/logs# ls -lh
total 3.3M
-rw-r--r-- 1 www-data www-data 3.3M Dec  3 15:35 2024-12-03-163132-update.txt
```

And the log file has the whole process. We're good to merge!

**Pull request merged**